File: Easy/977. Squares of a Sorted Array/solution.py
class Solution:
    def sortedSquares(self, A: List[int]) -> List[int]:
        # def square(num):
        #     return num * num
        
        print(A)
        # unsortedSquares = map(square, A)
        # print(sorted(unsortedSquares))
        
        # for item in unsortedSquares:
        #     print(item)
            
        # sortedSquares = unsortedSquares.sort()
        # for item in sortedSquares:
        #     print(item)
            
        # return sorted(map(square, A))
        return sorted(num*num for num in A)
      

File: Easy/977. Squares of a Sorted Array/solution.js
/**Given an array of integers A sorted in non-decreasing order, return an array of the squares of each number, also in sorted non-decreasing order.

 

Example 1:

Input: [-4,-1,0,3,10]
Output: [0,1,9,16,100]
 */

 /**
 * @param {number[]} A
 * @return {number[]}
 */
var sortedSquares = function(A) {
  return A.map(num => num * num).sort((a,b) => a-b);
};